src/NHibernate/Impl/ExpressionProcessor.cs

```diff
@@ -494,26 +472,18 @@ private static System.Type FindMemberType(Expression expression)
 				return FindMemberType(unwrapExpression);
 			}
 
-			var methodCallExpression = expression as MethodCallExpression;
-			if (methodCallExpression != null)
-			{
-				return methodCallExpression.Method.ReturnType;
-			}
-
-			if (expression is BinaryExpression || expression is UnaryExpression)
-				return expression.Type;
-
-			throw new ArgumentException("Could not determine member type from " + expression, nameof(expression));
+			return expression.Type;
```

Comment:
The previous code would have throw if the expression is not a binary, unary or member expression. Why do we not need this check? Is it because it would be already somehow be checked by the `FindMemberProjection` call that is currently done prior to calling `FindMemberType`? It does not look to me as being the case.

Reply:
I just thought it's not that important to have strict check inside this private method as long `FindMemberProjection` is able to parse expression.. But maybe not - I restored exception